### PR TITLE
[#149] Handling delivery profiles

### PR DESCRIPTION
**Pull request by jeniaefimov**

### CODE REVIEW

File: app/services/spree_shopify_importer/data_savers/tax_categories/saver.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module SpreeShopifyImporter
+  module DataSavers
+    module TaxCategories
+      class Saver
```
Comment (joaray) [inline app/services/spree_shopify_importer/data_savers/tax_categories/saver.rb]: what about keeping convention and tax_category_creator.rb?

File: app/services/spree_shopify_importer/importers/variant_importer.rb [hunk 2]
```diff
@@ -12,20 +12,22 @@ def import!
...
+          creator.new(data_feed, spree_product, shopify_image).create!
         else
-          updater.new(data_feed, spree_object, @spree_product, @shopify_image).update!
+          updater.new(data_feed, spree_object, spree_product, shopify_image).update!
         end
       end
 
       private
 
+      attr_reader :resource, :parent_feed, :spree_product, :shopify_image
```
Comment (kgorazd) [inline app/services/spree_shopify_importer/importers/variant_importer.rb]: we should talk about this first  and decide about convention

Comment (kgorazd) [reply app/services/spree_shopify_importer/importers/variant_importer.rb]: It this coherent with the rest of the code ?

Comment (jeniaefimov) [reply app/services/spree_shopify_importer/importers/variant_importer.rb]: I'm ready to discuss

Comment (kgorazd) [reply app/services/spree_shopify_importer/importers/variant_importer.rb]: Thing is if this convention is used in the rest of the code - it is ok to do that here. If it is not - double standards will create a mess.
If we want to use it. Then in separate PR we can do refactor..

Comment (jeniaefimov) [reply app/services/spree_shopify_importer/importers/variant_importer.rb]: let me check, and I'll be back with comment

Comment (jeniaefimov) [reply app/services/spree_shopify_importer/importers/variant_importer.rb]: Common practice in our codebase is to write `@instance_variable.method_call`, but as for me the way with `attr_reader` is better, and I can do a refactoring when we'll have a time.
I'll try to put a few "pros" of `attr_reader` way below.

Comment (kgorazd) [reply app/services/spree_shopify_importer/importers/variant_importer.rb]: Convention > personal preference.
I am not opposed to private attr_reader, they are used in other our projects, but in this project not really. We can change that someday. We should not use many different styles/conventions.

Comment (jeniaefimov) [reply app/services/spree_shopify_importer/importers/variant_importer.rb]: To be honest I agree with you, I was thinking a bit why I'm writing services in this way and I found that it's just a personal preference or may be just a habit (even from Java time, where all operations with object state carried out using `getters` and `setters`). I'll change a code to follow the convention.
Thanks for discussing.

Comment (kgorazd) [reply app/services/spree_shopify_importer/importers/variant_importer.rb]: We can change that convention - in many projects we are using attr_readers. Just in separate PRs and for whole namespaces at once. Not a priority right now, though.

Comment (jeniaefimov) [reply app/services/spree_shopify_importer/importers/variant_importer.rb]: got it.
Already changed the code.

File: app/services/spree_shopify_importer/data_savers/tax_categories/saver.rb (hunk 1 above)
Comment (jeniaefimov) [reply app/services/spree_shopify_importer/data_savers/tax_categories/saver.rb]: oh, sure, I'll change it to creator. But do you think that we need to repeat the name in the module and in the class name?
I mean module `TaxCategories` and class `TaxCategoryCreator` vs module `TaxCategories`  and class `Creator`?
I think that we created a module in order to understand the context and we don't need repetition.

Comment (kgorazd) [reply app/services/spree_shopify_importer/data_savers/tax_categories/saver.rb]: Oh I would go with something like this 
SpreeShopifyImporter::DataSavers::TaxCategory::Create
SpreeShopifyImporter::DataSavers::TaxCategory::Update
but again if other services are called SpreeShopifyImporter::DataSavers::Itemss::ItemCreator we should stick with it, and eventually do a global refactor.

Comment (kgorazd) [reply app/services/spree_shopify_importer/data_savers/tax_categories/saver.rb]: So yes it doesn't make much sense to repeat it, but this is some kind of convention in this code and we should stick with it, or change convention to the new one - but not mix them under one namespace.

Comment (jeniaefimov) [reply app/services/spree_shopify_importer/data_savers/tax_categories/saver.rb]: @kgorazd  can I create a separate PR and change the convention, for example today?
I don't like this repetition.

Comment (jeniaefimov) [reply app/services/spree_shopify_importer/data_savers/tax_categories/saver.rb]: so, I'll change the name to `tax_category_creator.rb`but may be later, let's discuss how to change it.

**Comment by jeniaefimov** [comment]
Screenshot 2019-12-02 at 10 43 15

@joaray I'll answer to your comment here.
As for parser, I don't need to prepare any data after I fetched it from shopify, it comes in good format and I can immediately save it to tax category, if I missed something please explain me once again.

About `delivery profile importer`, I think not a lot is going on there.

File: app/services/spree_shopify_importer/data_savers/tax_categories/saver.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        def initialize(delivery_profile)
+          @delivery_profile = delivery_profile
+        end
+
+        def call
+          Spree::TaxCategory.where(name: delivery_profile.name, is_default: delivery_profile.default).first_or_create!
```
[the remark this answers is not in the document]
Comment (jeniaefimov) [reply app/services/spree_shopify_importer/data_savers/tax_categories/saver.rb]: answered above

File: app/services/spree_shopify_importer/importers/delivery_profile_importer.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+module SpreeShopifyImporter
+  module Importers
+    class DeliveryProfileImporter
+      def initialize(spree_product, shopify_product)
+        @spree_product = spree_product
+        @shopify_product = shopify_product
+      end
+
+      def call
```
[the remark this answers is not in the document]
Comment (jeniaefimov) [reply app/services/spree_shopify_importer/importers/delivery_profile_importer.rb]: answered above. Import delivery profile and transform to tax_category.

**Pull request merged**